src/ILVerify/tests/ILTests/FtnTests.il

```diff
@@ -0,0 +1,398 @@
+    {
+        // (object c)
+        // var a = new System.Action(c.InstanceMethod);
+
+        ldarg.0
+        dup
+        ldvirtftn   instance void TestClass::InstanceMethod()
+        newobj      instance void [System.Runtime]System.Action::.ctor(object, native int)
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.RefConstraint_Valid() cil managed
+    {
+        // var c = new GenericTestClass<TestClass>();
+
+        newobj      instance void class GenericTestClass`1<class TestClass>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.IComparableConstraint_Valid() cil managed
+    {
+        // var c = new GenericIComparable<int>();
+
+        newobj      instance void class GenericIComparable`1<int32>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.CtorConstraint_Valid() cil managed
+    {
+        // GenericCtor<object> c = new GenericCtor<object>();
+
+        newobj      instance void class GenericCtor`1<class [System.Runtime]System.Object>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.ValueTypeConstraint_Valid() cil managed
+    {
+        // var c = new GenericValueType<int>();
+
+        newobj      instance void class GenericValueType`1<int32>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.RefGenericForRefConstraint_Valid<class T>() cil managed
+    {
+        // var c = new GenericTestClass`1<T>();
+
+        newobj      instance void class GenericTestClass`1<!!T>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.ValueTypeGenericForValueTypeConstraint_Valid<valuetype T>() cil managed
+    {
+        // var c = new GenericValueType`1<T>();
+
+        newobj      instance void class GenericValueType`1<!!T>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.CtorGenericForCtorConstraint_Valid<.ctor T>() cil managed
+    {
+        // var c = new GenericCtor`1<T>();
+
+        newobj      instance void class GenericCtor`1<!!T>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.ValueTypeGenericForCtorConstraint_Valid<valuetype T>() cil managed
+    {
+        // var c = new GenericCtor`1<T>();
+
+        newobj      instance void class GenericCtor`1<!!T>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.IntForRefConstraint_Invalid_UnsatisfiedMethodParentInst() cil managed
+    {
+        // GenericTestClass<int> c = new GenericTestClass<int>();
+
+        newobj      instance void class GenericTestClass`1<int32>::.ctor()
+        pop
+        ret
+    }
+
+    .method static public hidebysig void NewObj.ClassForIComparableConstraint_Invalid_UnsatisfiedMethodParentInst() cil managed
```

Comment:
It may not be require to have coverage for as many variants of valid or invalid constraints here. The full matrix is covered by the constraints unit tests.

Reply:
True, I will also have a look at this when adding the mentioned test coverage on Monday.